src/runtime_src/core/pcie/tools/xbmgmt/cmd_config.cpp

```diff
@@ -224,25 +242,40 @@ static int show(int argc, char *argv[])
     return 0;
 }
 
-static void updateDevConf(std::shared_ptr<pcidev::pci_device>& dev,
-    std::string lvl)
+static void updateDevConf(pcidev::pci_device *dev,
+    const std::string lvl, configType config_type)
 {
     std::string errmsg;
-    dev->sysfs_put("icap", "sec_level", errmsg, lvl);
-    if (!errmsg.empty()) {
-        std::cout << "Failed to set security level for " << dev->sysfs_name
-            << std::endl;
-        std::cout << "See dmesg log for details" << std::endl;
+
+    switch(config_type) {
+    case CONFIG_SECURITY:
+        dev->sysfs_put("icap", "sec_level", errmsg, lvl);
+        if (!errmsg.empty()) {
+            std::cout << "Failed to set security level for " <<
+                dev->sysfs_name << "\n";
+            std::cout << "See dmesg log for details" << std::endl;
+        }
+        break;
+    case CONFIG_CLK_SCALING:
+        dev->sysfs_put("xmc", "scaling_enabled", errmsg, lvl);
+        if (!errmsg.empty()) {
+            std::cout << "Failed to update clk scaling status for " <<
+                dev->sysfs_name << "\n";
+            std::cout << "See dmesg log for details" << std::endl;
+        }
+        break;
     }
 }
 
 static int device(int argc, char *argv[])
 {
     unsigned int index = UINT_MAX;
     std::string lvl;
+    configType config_type = (configType)-1;
```

Comment:
To fix below errors, defined config_type like above.
1. error: config_type may be used uninitialized in this function
2. error: invalid conversion from int to configType [-fpermissive]